xcube/core/gen/iproc.py

```diff
@@ -120,6 +120,17 @@ def get_extra_vars(self, dataset: xr.Dataset) -> Optional[Collection[str]]:
         """
         return None
 
+    def get_spatial_subset(self, dataset: xr.Dataset, dst_region: Tuple[float, float, float, float]) -> xr.Dataset:
```

Comment:
Additional, important notes:

* `get_spatial_subset` should be abstract in `InputProcessor`.
* This `get_spatial_subset` implementation should go into `XYInputProcessor`.
* This `get_spatial_subset` implementation for `DefaultInputProcessor` is much simpler and much faster, because we rectified coordinates already.

Reply:
Maybe  `get_spatial_subset` should not be a new method at all `InputProcessor`. There are already too much - adding a lot of responsibility for developers to implement a new one.
We have already the `pre_process` method for pre-processing, and subsetting *is a* pre-processing.
It may be better if we move a `get_spatial_subset` into `xcube.core.geom` and use it in the default implementation of `XYInputProcessor`.
That way, `get_spatial_subset` could be implemented in the fastest possible way and work for datasets 2-D `lon`, `lat` coordinates with 1-D `x`, `y` coordinates  (such as Sentinel-2) and for WGS84 inputs that have 1-D  `lon`, `lat` coordinates.